src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -124,6 +126,10 @@ public Builder dateTimeFormatter(FormatDateTimeFormatter dateTimeFormatter) {
         @Override
         public DateFieldMapper build(BuilderContext context) {
             fieldType.setOmitNorms(fieldType.omitNorms() && boost == 1.0f);
+            // index age check
+            if (dateTimeFormatter.equals(Defaults.DATE_TIME_FORMATTER) && context.indexCreatedVersion().before(Version.V_2_0_0)) {
```

Comment:
nit picky -  can we move this to use something similar to `getDateTimeFormatter` in TimestampFieldMapper ?